pkg/rtc/subscriptionmanager.go

```diff
@@ -50,14 +50,19 @@ type SubscriptionManagerParams struct {
 	OnTrackUnsubscribed func(subTrack types.SubscribedTrack)
 	OnSubscriptionError func(trackID livekit.TrackID)
 	Telemetry           telemetry.TelemetryService
+
+	SubscriptionLimitVideo, SubscriptionLimitAudio int32
 }
 
 // SubscriptionManager manages a participant's subscriptions
 type SubscriptionManager struct {
 	params        SubscriptionManagerParams
 	lock          sync.RWMutex
 	subscriptions map[livekit.TrackID]*trackSubscription
-	subscribedTo  map[livekit.ParticipantID]map[livekit.TrackID]struct{}
+
+	subscribedVideoCount, subscribedAudioCount atomic.Int32
```

Comment:
just thinking out aloud, should we have limits per track source?

Reply:
Do you mean camera or screen share? I can't image the scenario to limit at source level now

Reply:
yeah, I was thinking source based. We can do it later.

I was thinking about a case of reaching video limit and somebody starts a screen share. I am guessing users would want to receive that. So, having a different limit would help with that. Also, there is not going to be a lot of parallel screen shares. So, that can have a low limit. Of course, we can added it later.

Reply:
Sounds like a priority, makes sense to have screen share as preferred.